leakcanary-object-watcher-android/src/main/java/leakcanary/internal/ServiceDestroyWatcher.kt

```diff
@@ -0,0 +1,125 @@
+package leakcanary.internal
+
+import android.annotation.SuppressLint
+import android.app.Service
+import android.os.Build
+import android.os.Handler
+import android.os.IBinder
+import android.util.ArrayMap
+import leakcanary.AppWatcher
+import leakcanary.ObjectWatcher
+import shark.SharkLog
+import java.lang.reflect.Proxy
+
+internal class ServiceDestroyWatcher private constructor(
+  private val objectWatcher: ObjectWatcher,
+  private val configProvider: () -> AppWatcher.Config
+) {
+
+  private val mTmpServices = mutableMapOf<IBinder, Service>()
```

Comment:
We likely want a map<IBinder, Service>, and maybe even a weak hashmap (ie idk if we should risk retaining the IBinder either)

Reply:
Followed up  here: https://github.com/square/leakcanary/pull/2029